### PR TITLE
Initial support for ADC hal

### PR DESCRIPTION
**Pull request by duynguyenxa**
This PR add ADC driver to hal_renesas

### CODE REVIEW

**Comment by ydamigos** [comment]
@thaoluonguw Could you please check it? I can't add you as a reviewer.

File: zephyr/ra/ra_cfg/fsp_cfg/r_adc_cfg.h
Comment (soburi) [inline zephyr/ra/ra_cfg/fsp_cfg/r_adc_cfg.h:1]: Please clarify the license and author.

**Review by soburi — CHANGES_REQUESTED** [review]

**Review by ydamigos — APPROVED** [review]

File: zephyr/ra/ra_cfg/fsp_cfg/r_adc_cfg.h
Comment (thaoluonguw) [reply zephyr/ra/ra_cfg/fsp_cfg/r_adc_cfg.h]: We updated. Thanks for your feedback.

**Review by thaoluonguw — APPROVED** [review]

**Review by soburi — APPROVED** [review]
The zephyr part is OK.
The FSP part probably contains code that is unreleased, so it's difficult for me to judge. (I think it's probably OK though.)

**Comment by KhiemNguyenT** [comment]
> The zephyr part is OK. The FSP part probably contains code that is unreleased, so it's difficult for me to judge. (I think it's probably OK though.)

@duynguyenxa , please give an explanation for the diff.

**Comment by duynguyenxa** [comment]
> The zephyr part is OK. The FSP part probably contains code that is unreleased, so it's difficult for me to judge. (I think it's probably OK though.)

@soburi , this is a snapshot of FSP at release 5.3.0, are compare with 5.3.0 or the latest version on Renesas github (5.4.0)

**Comment by soburi** [comment]
@duynguyenxa @KhiemNguyenT 
> > The zephyr part is OK. The FSP part probably contains code that is unreleased, so it's difficult for me to judge. (I think it's probably OK though.)
> 
> @soburi , this is a snapshot of FSP at release 5.3.0, are you compare with FSP v5.3.0 or the latest version on Renesas github (v5.4.0)

If that is the case, there will be a discrepancy because the currently installed RA8M1 and SCI-B drivers are v5.3.0. This may not be a problem in general, but it will be a chimera, which is not desirable from a management perspective.
My only concern is the increased management effort, which is not a problem as long as it does not create a huge burden.

**Comment by KhiemNguyenT** [comment]
> @duynguyenxa @KhiemNguyenT
> 
> > > The zephyr part is OK. The FSP part probably contains code that is unreleased, so it's difficult for me to judge. (I think it's probably OK though.)
> > 
> > 
> > @soburi , this is a snapshot of FSP at release 5.3.0, are you compare with FSP v5.3.0 or the latest version on Renesas github (v5.4.0)
> 
> If that is the case, there will be a discrepancy because the currently installed RA8M1 and SCI-B drivers are v5.3.0. This may not be a problem in general, but it will be a chimera, which is not desirable from a management perspective. My only concern is the increased management effort, which is not a problem as long as it does not create a huge burden.

@soburi , this is the open-source development, then we may relax the policy. In general, we try to keep sync with latest FSP HAL, but it's not the mandatory condition to accept the PRs. e.g. we will not wait for new FSP release in order to start some development.
In case there's a diff which leads to some issues, a PR will be submitted. Does it make sense ?

**Comment by soburi** [comment]
@KhiemNguyenT 

Yes , I think it make a sense.

Rather than the need to synchronize, I think it is necessary to manage the differences with the upstream (we need to know and manage what version we are used, and what chagen we maked).
The simple problem is that when an unpublished change is presented here, it can be hard to tell whether it's something that will merge upstream in the future, or whether it's something specific to the Zephyr repository.
I think that if a PR is made to upstream, this will be easier to manage.
I'm not to say that you shouldn't use the latest version.

**Comment by ydamigos** [comment]
@KhiemNguyenT PR has three approvals. Should you merge it? If yes, we should remove draft label from PR #76127.

**Pull request merged**
